bitcoin/src/consensus/encode.rs

```diff
@@ -126,6 +168,19 @@ pub fn deserialize<T: Decodable>(data: &[u8]) -> Result<T, Error> {
     }
 }
 
+/// Deserialize any decodable type from a hex string, will error if said deserialization
+/// doesn't consume the entire vector.
+pub fn deserialize_hex<T: Decodable>(hex: &str) -> Result<T, FromHexError> {
+    let mut iter = HexToBytesIter::new(hex)?; // iter implements `io::Read` without allocation.
```

Comment:
If you wrap this in `serde::IterReader` it should resolve the `io` issue. You need to make it `pub(crate)` and move it out of the module of course.

Reply:
And you most likely have to do it because of `BufRead`.

Reply:
Cool, thanks for the tip.

Reply:
Now done.